File: padroesEstruturais/padrao-de-projeto-adapter/java/src/main/java/Partida.java
public class Partida {

    IDano dano = null;
    DanoAdapter danoAdapter = null;

    public Partida() {
        this.dano = new DanoNumerico();
        this.danoAdapter = new DanoAdapter(this.dano);
    }

    public Partida setDano(float avaliacao) {
        this.dano.setDano(avaliacao);
        this.danoAdapter.salvarDano();
        return this;
    }

    public String getDanoPorcentagem() {
        return danoAdapter.getDano();
    }

    public float getDanoNumerico() {
        return this.dano.getDano();
    }
}

File: padroesEstruturais/padrao-de-projeto-adapter/php/src/Adapter/Partida.php
<?php

namespace Adapter\Php\Adapter;

class Partida
{
    private ?IDano $dano = null;
    private ?DanoAdapter $danoAdapter = null;

    public function __construct()
    {
        $this->dano = new DanoNumerico();
        $this->danoAdapter = new DanoAdapter($this->dano);
    }

    public function setDano(float $dano): Partida
    {
        $this->dano->setDano($dano);
        $this->danoAdapter->salvarDano();
        return $this;
    }

    public function getDanoPorcentagem(): String
    {
        return $this->danoAdapter->getDano();
    }

    public function getDanoNumerico(): float
    {
        return $this->dano->getDano();
    }
}